specification/elasticsan/resource-manager/Microsoft.ElasticSan/preview/2021-11-20-preview/elasticsan.json

```diff
@@ -0,0 +1,2658 @@
+      "description": "The API version to use for this operation.",
+      "minLength": 1
+    },
+    "ResourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the resource group. The name is case insensitive.",
+      "pattern": "^[-\\w\\._]*[0-9A-Za-z]$",
+      "minLength": 1,
+      "maxLength": 90,
+      "x-ms-parameter-location": "method"
+    },
+    "ElasticSanNameParameter": {
+      "name": "elasticSanName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the ElasticSan.",
+      "pattern": "^[A-Za-z0-9]+((-|_)[a-z0-9A-Z]+)*$",
+      "minLength": 3,
+      "maxLength": 24,
+      "x-ms-parameter-location": "method"
+    },
+    "VolumeGroupNameParameter": {
+      "name": "volumeGroupName",
+      "type": "string",
+      "required": true,
+      "in": "path",
+      "description": "The name of the VolumeGroup.",
+      "pattern": "^[A-Za-z0-9]+((-|_)[a-z0-9A-Z]+)*$",
+      "minLength": 3,
+      "maxLength": 63,
+      "x-ms-parameter-location": "method"
+    },
+    "VolumeNameParameter": {
+      "name": "volumeName",
+      "type": "string",
+      "required": true,
+      "in": "path",
+      "description": "The name of the Volume.",
+      "pattern": "^[a-z0-9]+(-[a-z0-9A-Z]+)*$",
+      "minLength": 3,
+      "maxLength": 63,
+      "x-ms-parameter-location": "method"
+    },
+    "ForceDeleteSnapshotsParameter": {
+      "name": "x-ms-delete-snapshots",
+      "x-ms-client-name": "forceDeleteSnapshots",
+      "description": "Required if the Volume has associated snapshots",
+      "x-ms-parameter-location": "method",
+      "in": "header",
+      "required": false,
+      "type": "boolean"
+    },
+    "ForceDeleteVolumesSnapshotsParameter": {
+      "name": "x-ms-delete-volumes-snapshots",
+      "x-ms-client-name": "forceDeleteVolumesAndSnapshots",
+      "description": "Required if the Volume Group has associated volumes and snapshots",
+      "x-ms-parameter-location": "method",
+      "in": "header",
+      "required": false,
+      "type": "boolean"
+    },
+    "ForceDeleteVolumeGroupsParameter": {
+      "name": "x-ms-delete-volumegroups",
+      "x-ms-client-name": "forceDeleteVolumeGroups",
+      "description": "Required if the San has associated Volume Groups",
+      "x-ms-parameter-location": "method",
+      "in": "header",
+      "required": false,
+      "type": "boolean"
+    },
+    "SnapshotNameParameter": {
+      "name": "snapshotName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the snapshot within the given subscription, resource and volume group.",
+      "pattern": "^[a-z0-9]+(?:[._-][a-z0-9]+)*$",
+      "minLength": 1,
+      "maxLength": 80,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "ElasticSanOperationDisplay": {
```

Comment:
>ElasticSanOperationDisplay [](http://example.com/codeflow?start=5&length=26)

Are these properties set by your RP? If yes, mark your properties as readonly: true, this indicates that customers won't be able to set these values.

Reply:
@jorgecotillo - these is a required property and it is used only in list operations (so that implies readonly, correct) ? When I tried to set it to "read-only", the validator complained that it is a required property

`{
  "type": "Error",
  "code": "RequiredReadOnlyProperties",
  "message": "Property 'display' is a required property. It should not be marked as 'readonly'.",
  "id": "R2056",
  "validationCategory": "SDKViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///C:/Users/pprahalad.FAREAST/azrapi/specification/elastiCSAN/resource-manager/Microsoft.ElasticSan/preview/2021-11-20-preview/elasticsan.json:1549:8 ($.definitions.ElasticSanRPOperation.properties.display)"
  ],
  "jsonref": "file:///C:/Users/pprahalad.FAREAST/azrapi/specification/elastiCSAN/resource-manager/Microsoft.ElasticSan/preview/2021-11-20-preview/elasticsan.json:1549:8 ($.definitions.ElasticSanRPOperation.properties.display)",
  "json-path": "file:///C:/Users/pprahalad.FAREAST/azrapi/specification/elastiCSAN/resource-manager/Microsoft.ElasticSan/preview/2021-11-20-preview/elasticsan.json:1549:8 ($.definitions.ElasticSanRPOperation.properties.display)"
`

Reply:
Updated.